src/ILCompiler.Compiler/src/CppCodeGen/ILToCppImporter.cs

```diff
@@ -477,7 +482,29 @@ private void EndImportingInstruction()
         public void Compile(CppMethodCodeNode methodCodeNodeNeedingCode)
         {
             FindBasicBlocks();
+            for (int i = 0; i < methodCodeNodeNeedingCode.Method.Signature.Length; i++)
+            {
+                var parameterType = methodCodeNodeNeedingCode.Method.Signature[i];
+                //if ((parameterType.IsTypeDefinition || !parameterType.HasSameTypeDefinition(_compilation.NodeFactory.ArrayOfTClass)))
```

Comment:
If we not longer need this, let's just remove the commented code and the braces below around AddTypeReference